parsers/sml.c

```diff
@@ -31,7 +32,8 @@ typedef enum {
 	K_SIGNATURE,
 	K_STRUCTURE,
 	K_TYPE,
-	K_VAL
+	K_VAL,
+	K_LAST_ELT = K_VAL
```

Comment:
I'd use something like `K_COUNT /* must be last */` and simply keep it last, to avoid having to update it when adding a kind.  But that's mostly style, so it's your call.